config-service-impl/src/test/java/org/hypertrace/config/service/store/DocumentConfigStoreTest.java

```diff
@@ -162,68 +163,63 @@ void writeAllConfigs() throws IOException {
     ConfigResourceContext resourceContext1 = getConfigResourceContext("context-1");
     ConfigResourceContext resourceContext2 = getConfigResourceContext("context-2");
     long updateTime = 1234;
-    CloseableIterator<Document> firstGetResult =
+    CloseableIterator<Document> getResult =
         new CloseableIteratorImpl(
             List.of(
                 getConfigDocument(
-                    resourceContext1.getContext(),
-                    CONFIG_VERSION,
-                    config1,
-                    TIMESTAMP1,
-                    TIMESTAMP1)));
-
-    CloseableIterator<Document> secondGetResult =
-        new CloseableIteratorImpl(
-            List.of(
+                    resourceContext1.getContext(), CONFIG_VERSION, config1, TIMESTAMP1, TIMESTAMP1),
                 getConfigDocument(
                     resourceContext2.getContext(),
                     CONFIG_VERSION,
                     config2,
                     TIMESTAMP2,
                     TIMESTAMP2)));
-    when(collection.search(any(Query.class))).thenReturn(firstGetResult, secondGetResult);
+
+    when(collection.search(any(Query.class))).thenReturn(getResult);
+    when(collection.bulkUpsert(any())).thenReturn(true);
     when(this.mockClock.millis()).thenReturn(updateTime);
     List<UpsertedConfig> upsertedConfigs =
         // Swap configs between contexts as an update
         configStore.writeAllConfigs(
             ImmutableMap.of(resourceContext1, config2, resourceContext2, config1), USER_ID);
 
-    assertEquals(
-        List.of(
+    assertEquals(2, upsertedConfigs.size());
```

Comment:
Do we no longer have a guarantee of order? I wouldn't expect the api contract to change for an impl detail.